src/htm/regions/TestNode.cpp

```diff
@@ -35,17 +35,24 @@
 
 namespace htm {
 
-TestNode::TestNode(const ValueMap &params, Region *region)
+TestNode::TestNode(const ValueMap &par, Region *region)
     : RegionImpl(region), computeCallback_(nullptr), nodeCount_(1)
 {
+  outputElementCount_ = 0;
+  spec_.reset(createSpec());
+  ValueMap params = ValidateParameters(par, spec_.get());
```

Comment:
is this the new usage? From params, obtain new, validated params by calling `ValidateParameters()`? Couldn't the validation be done implicitly?

Reply:
> Couldn't the validation be done implicitly?

Originally this was performed in the factory before creating the region. but the region implementation had less control.  I struggled with this for quite a while.  I may yet go back to performing it in the factory before creating the region.